--- src/main/java/io/github/dsheirer/channel/metadata/NowPlayingPanel.java
+++ src/main/java/io/github/dsheirer/channel/metadata/NowPlayingPanel.java
@@ -87,11 +87,16 @@
     }

     public void setComponents(javax.swing.JComponent spectralPanel, javax.swing.JComponent broadcastStatusPanel, javax.swing.JComponent resourceStatusPanel) {
+        boolean initialized = (mSpectralPanel != null);
         mSpectralPanel = spectralPanel;
         mBroadcastStatusPanel = broadcastStatusPanel;
         mResourceStatusPanel = resourceStatusPanel;

-        setupWidgets();
+        if (!initialized) {
+            setupWidgets();
+        } else {
+            mWidgetContainer.ensureComponentInWidget("spectrum");
+        }
     }
