--- src/main/java/io/github/dsheirer/gui/widget/Widget.java
+++ src/main/java/io/github/dsheirer/gui/widget/Widget.java
@@ -113,6 +113,14 @@
         return mHeaderPanel;
     }

+    public void ensureContentComponentParent() {
+        if (mContentComponent != null && mContentComponent.getParent() != this) {
+            add(mContentComponent, "grow, wrap", 1);
+            revalidate();
+            repaint();
+        }
+    }
+
     public void setDragging(boolean dragging) {
         if (dragging) {
             mHeaderPanel.setBackground(UIManager.getColor("Component.focusColor"));
